de.fu_berlin.inf.dpp.intellij/src/de/fu_berlin/inf/dpp/intellij/ui/views/buttons/ConnectButton.java

```diff
@@ -123,54 +142,202 @@ private void createAddAccountMenuItem() {
-                        .showError("Account was not created.", "Error");
+                    return;
                 }
+
                 createMenuItems();
+
+                askToConnectToAccount(account);
             }
         });
     }
 
+    private void askToConnectToAccount(@NotNull XMPPAccount account){
+        try {
+            Integer option = SafeDialogUtils.showYesNoDialog(
+                Messages.ConnectButton_connect_to_new_account_message,
+                Messages.ConnectButton_connect_to_new_account_title);
+
+            if (option == com.intellij.openapi.ui.Messages.YES){
+                connectAction.executeWithUser(
+                    account.getUsername() + USERID_SEPARATOR
+                        + account.getDomain());
+            }
+
+
+        } catch (IllegalAWTContextException e) {
+            LOG.error("Account creation failed.", e);
+
+            SafeDialogUtils.showError(
+                "There was an error creating the account.\nDetails: "
+                    + e.getMessage(),
+                "Account creation failed");
+        }
+    }
+
     /**
-     * Asks for Name, Password and domain and server for a new XMPP account.
+     * Asks for user id (name@domain), password, and server for a new XMPP
+     * account.
+     *
+     * @return an <code>XMPPAccount</code> with the values entered by the user
+     *         or <code>null</code> if the creation was canceled by the user or
+     *         the user entered at least one illegal value
      */
-    protected XMPPAccount createNewAccount() {
-        final String userID = SafeDialogUtils.showInputDialog(
-            "Your User-ID, e.g. user@saros-con.imp.fu-berlin.de", "", "Login");
-        if (userID.isEmpty()) {
+    private XMPPAccount createNewAccount() {
+        final String userID;
+
+        try {
+            userID = SafeDialogUtils.showInputDialog(
+                "Your User-ID, e.g. user@saros-con.imp.fu-berlin.de",
+                "", "Login");
+
+        } catch (IllegalAWTContextException e) {
+            LOG.error("Account creation failed.", e);
+
+            SafeDialogUtils.showError(
+                "There was an error creating the account.\nDetails: "
+                    + e.getMessage(), "Account creation failed");
+
             return null;
         }
-        if (!userID.contains("@")) {
-            SafeDialogUtils.showError("No @ found in the ID!", "Error");
+
+        if (userID == null) {
+            LOG.debug("Account creation canceled by user during user id"
+                + " entry.");
+
             return null;
         }
 
-        String[] fields = userID.split(USERID_SEPARATOR);
-        String username = fields[0];
-        String domain = fields[1];
+        JID jid = new JID(userID);
+
+        if (!jid.isValid() || jid.getName().isEmpty()) {
+            SafeDialogUtils.showError("Entered user id is not valid.",
+                "Account creation aborted");
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Define a constant instead of duplicating this literal "Account creation aborted" 3 times. ([squid:S1192](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AS1192))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=238&issues=AWYa7GOkvOWp7Yd62c1J&open=AWYa7GOkvOWp7Yd62c1J)

[ISSUE_KEY]: AWYa7GOkvOWp7Yd62c1J